api-admin/src/main/java/bio/terra/pearl/api/admin/service/forms/SurveyExtService.java

```diff
@@ -27,6 +29,34 @@ public SurveyExtService(
     this.studyEnvironmentSurveyService = studyEnvironmentSurveyService;
   }
 
+  public Optional<Survey> get(
+      String portalShortcode, String stableId, int version, AdminUser adminUser) {
+    Portal portal = authUtilService.authUserToPortal(adminUser, portalShortcode);
+    Optional<Survey> surveyOpt = surveyService.findByStableId(stableId, version);
+    if (surveyOpt.isEmpty()) {
+      return surveyOpt;
+    }
+    Survey survey = surveyOpt.get();
+    if (!portal.getId().equals(survey.getPortalId())) {
```

Comment:
It wasn't immediately obvious to me exactly what this was guarding. Seemed odd to have a "permission denied" error so far down in the function.

I think it's worth extracting the code that protects against mismatched portal/survey identifiers. Maybe make `surveyService.findByStableId` also require the `portalShortcode` and do the check there. Or introduce a `authUtilService.authUserToSurvey` function. Otherwise, it'll be too easy to forget to make this check.